### PR TITLE
Create event-bus-knative helm chart

### PR DESCRIPTION
**Pull request by montaro**
See also #2279

### CODE REVIEW

File: resources/event-bus/templates/tests/test-e2e-tester.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ metadata:
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: test-core-event-bus-subs
+  name: -test-event-bus-subs
```
Comment (abbi-gaurav) [inline resources/event-bus/templates/tests/test-e2e-tester.yaml]: Is this the convention to follow for charts outside core? (name starting with `-`).
Same question applied for others/

Comment (montaro) [reply resources/event-bus/templates/tests/test-e2e-tester.yaml]: Thanks for the catch, no, it's by mistake

File: resources/event-bus/values.yaml [hunk 2]
```diff
@@ -2,10 +2,17 @@
...
+    path: eu.gcr.io/kyma-project
+  event_bus:
+    dir: develop/
+    version: ab9c73dd
+  event_bus_tests:
+    dir: develop/
+    version: f01c8848
   natsStreaming:
     url: "http://nats-streaming.natss:4222"
     clusterID: "kyma-nats-streaming"
```
Comment (radufa) [inline resources/event-bus/values.yaml]: clusterID, as discussed, should be a global variable, cause for Knative deployment its value is: "knative-nats-streaming"

File: installation/resources/installer-cr-cluster-xip-io.yaml.tpl [hunk 3]
```diff
@@ -24,6 +24,8 @@ spec:
       namespace: "kyma-system"
     - name: "nats-streaming"
       namespace: "natss"
+    - name: "event-bus"
+      namespace: "kyma-system"
```
Comment (radufa) [inline installation/resources/installer-cr-cluster-xip-io.yaml.tpl]: Why "event-bus" should be installed before core? I think that "event-bus" should be the first component installed immediately after core

File: installation/resources/installer-cr-cluster.yaml.tpl [hunk 4]
```diff
@@ -16,6 +16,8 @@ spec:
       namespace: "natss"
     - name: "knative"
       namespace: "istio-system" # KNative comes with namespaces hardcoded so this one is only for installer
+    - name: "event-bus"
+      namespace: "kyma-system"
```
Comment (radufa) [inline installation/resources/installer-cr-cluster.yaml.tpl]: The same, see the previous comment.

File: installation/resources/installer-cr.yaml.tpl [hunk 5]
```diff
@@ -16,6 +16,8 @@ spec:
       namespace: "natss"
     - name: "knative"
       namespace: "istio-system" # KNative comes with namespaces hardcoded so this one is only for installer
+    - name: "event-bus"
+      namespace: "kyma-system"
```
Comment (radufa) [inline installation/resources/installer-cr.yaml.tpl]: The same, see the previous comment.

File: resources/event-bus/charts/publish-knative/Chart.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+description: Publish Knative Helm chart.
+name: publish-knative
+version: 0.1.0
```
Comment (radufa) [inline resources/event-bus/charts/publish-knative/Chart.yaml]: Please add new line

File: resources/event-bus/requirements.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+dependencies:
+  - name: publish
+    condition: global.eventBus
+  - name: push
+    condition: global.eventBus
+  - name: sub-validator
+    condition: global.eventBus
+  - name: publish-knative
+    condition: global.knative
```
Comment (radufa) [inline resources/event-bus/requirements.yaml]: Please add a newline

File: resources/event-bus/values.yaml (hunk 2 above)
Comment (montaro) [reply resources/event-bus/values.yaml]: @radufa Yes, this is coming in the next commit.

File: installation/resources/installer-cr-cluster-xip-io.yaml.tpl (hunk 3 above)
Comment (montaro) [reply installation/resources/installer-cr-cluster-xip-io.yaml.tpl]: I'll move it down after `core` and test

File: resources/event-bus/requirements.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+dependencies:
+  - name: publish
+    condition: global.eventBus
```
Comment (abbi-gaurav) [inline resources/event-bus/requirements.yaml]: To make it more clear, should we consider calling it `global.not-knative`

File: components/application-operator/charts/application/values.yaml [hunk 9]
```diff
@@ -29,7 +29,7 @@ eventService:
       pullPolicy: IfNotPresent
     args:
       externalAPIPort: &eventServiceExternalAPIPort 8081
-      eventsTargetURL: http://core-publish.kyma-system.svc.cluster.local:8080/v1/events
+      eventsTargetURL: http://event-bus-publish.kyma-system.svc.cluster.local:8080/v1/events
```
Comment (abbi-gaurav) [inline components/application-operator/charts/application/values.yaml]: Does this change requires releasing the operator first?

File: resources/event-bus/requirements.yaml (hunk 8 above)
Comment (montaro) [reply resources/event-bus/requirements.yaml]: I would go for more like`kyma`/`built-in` but not the `not-`, what do you think?

File: components/application-operator/charts/application/values.yaml (hunk 9 above)
Comment (montaro) [reply components/application-operator/charts/application/values.yaml]: Yes! in a following PR

**Comment by radufa** [comment]
I've tested the PR on linux having Kyma installed standalone and on also top of Knative.

File: resources/event-bus/requirements.yaml (hunk 8 above)
Comment (abbi-gaurav) [reply resources/event-bus/requirements.yaml]: `global.kyma-default` as knative is enabled with a feature flag.

**Comment by montaro** [comment]
test

**Pull request closed without merging**